### ISSUE TITLE
[Question] Is it possible compile CSharpCompilation to class object

### ISSUE THREAD

**Issue by Amitpnk** [npolyak/RoslynAssembly#1]
Thank your for article and code,

If Class A is in different file or different project lib, without string actual Class. Then is it possible to have reference in Class B

https://github.com/npolyak/RoslynAssembly/blob/master/Program.cs#L23

Regards,
Amit Naik

**Comment by npolyak** [comment]
Yes, I believe it is possible if class A is defined in a different assembly and the current project can access to that assembly. 
Note that the whole article is not valid for .NET core as it does not have netmodules. 
Regards

**Comment by Amitpnk** [comment]
thank you for reply, found similar article on .NET core.

Line 2: Instead of taking from path, is there any way to refer from dll

[reference link](https://docs.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/get-started/syntax-transformation#create-and-transform-trees)

![image](https://user-images.githubusercontent.com/24720047/139300578-58137e87-1c9b-4a82-9f5b-42a52731c367.png)

**Comment by npolyak** [comment]
I do not quite understand, you should create a reference from path - why cannot you do it?

**Comment by Amitpnk** [comment]
I want to refer from *.dll rather then path. Once we deploy, then it will be difficult get file path

**Comment by npolyak** [comment]
Yes, I understand, but I think you'll have to figure out the path with respect to the executable. You can create eg. a plugin folder and put all the plugin dlls there. I do not remember any other way to refer to a dll except by path.

**Comment by Amitpnk** [comment]
Thank you so much your help :)

**Comment by npolyak** [comment]
You are very welcome. BTW if a dll is loaded into the application, it is very easy to find its path via Reflection functionality - you need to find its Assembly and then the location of the Assembly.
